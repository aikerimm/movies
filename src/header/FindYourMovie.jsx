import { useCallback, useState } from 'react';
import { useNavigate, useParams, useSearchParams } from 'react-router-dom';
import Button from '../util/Button';
import React from 'react';

const FindYourMovie = () => {
  const params = useParams();
  const navigate = useNavigate();
  const [searchInput, setSearchInput] = useState(params.searchQuery);
  const [searchParams] = useSearchParams();

  const onSearchClick = useCallback(() => {
    navigate(`/search/${searchInput}?${searchParams.toString()}`);
  }, [navigate, searchParams, searchInput]);

  const handleInputChange = useCallback((e) => setSearchInput(e.target.value), []);

  const handleEnterOnSearch = useCallback(
    (e) => {
      if (e.key === 'Enter') {
        onSearchClick();
      }
    },
    [onSearchClick]
  );

  return (
    <div className='findYourMovie'>
      <p className='titleText fullWidth'>Find your movie</p>
      <div className='search'>
        <input
          type='text'
          placeholder='What do you want to watch?'
          className='searchInput roundedCorners'
          value={searchInput}
          onChange={handleInputChange}
          onKeyDown={handleEnterOnSearch}
        ></input>
        <Button value='search' type='submit' onClick={onSearchClick} />
      </div>
    </div>
  );
};

export default React.memo(FindYourMovie);
