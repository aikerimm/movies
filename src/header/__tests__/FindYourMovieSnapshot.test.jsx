/**
 * @jest-environment jsdom
 */

import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import FindYourMovie from '../FindYourMovie.jsx';

it('Find Your Movie renders correctly', () => {
  const tree = renderer.create(<BrowserRouter><FindYourMovie /></BrowserRouter>).toJSON();
  expect(tree).toMatchSnapshot();
});
