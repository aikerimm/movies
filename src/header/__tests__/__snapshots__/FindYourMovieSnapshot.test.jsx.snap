// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Find Your Movie renders correctly 1`] = `
<div
  className="findYourMovie"
>
  <p
    className="titleText fullWidth"
  >
    Find your movie
  </p>
  <div
    className="search"
  >
    <input
      className="searchInput roundedCorners"
      id="searchInput"
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="What do you want to watch?"
      type="text"
    />
    <input
      className="submitBtn roundedCorners"
      onClick={[Function]}
      type="button"
      value="search"
    />
  </div>
</div>
`;
